sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -42,28 +42,30 @@ case class Mode(
     this(child, 0, 0, Some(reverse))
   }
 
-  // Returns null for empty inputs
-  override def nullable: Boolean = true
-
-  override def dataType: DataType = child.dataType
-
-  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
-
   override def checkInputDataTypes(): TypeCheckResult = {
-    if (UnsafeRowUtils.isBinaryStable(child.dataType) || child.dataType.isInstanceOf[StringType]) {
+    if (UnsafeRowUtils.isBinaryStable(child.dataType) ||
+      !child.dataType.existsRecursively(f => f.isInstanceOf[MapType] &&
+        !UnsafeRowUtils.isBinaryStable(f))) {
       /*
         * The Mode class uses collation awareness logic to handle string data.
         * Complex types with collated fields are not yet supported.
        */
       // TODO: SPARK-48700: Mode expression for complex types (all collations)
       super.checkInputDataTypes()
     } else {
-      TypeCheckResult.TypeCheckFailure("The input to the function 'mode' was" +
-        " a type of binary-unstable type that is " +
-        s"not currently supported by ${prettyName}.")
+      TypeCheckResult.TypeCheckFailure("The input to the function 'mode' includes" +
+        " a map with keys and/or values which are not binary-stable. This is not yet" +
+        s"supported by ${prettyName}.")
```

Comment:
didn't we decide to include map support within this PR?

Reply:
I changed my mind about that. I will want to check a few things first, in a follow up.

Reply:
Up to you. If you really want Map in this PR, I can make that work. But that will definitely have to wait a few weeks. Merely tidying up what we have now might be able to be done ahead of schedule this or next week.

Reply:
of course, we can do MapType as a follow-up - let's just make sure to put a comment for this somewhere tho

and an appropriate test to catch this for now

Reply:
Will add!